Obfuscar/Obfuscator.cs

```diff
@@ -148,47 +154,83 @@ public void RunRules()
                     {
-                        TypeDefinition fieldType;
-                        if (field.IsPublic && (fieldType = field.FieldType.Resolve()) != null && fieldType.IsEnum)
-                            namesInXaml.Add(field.FieldType.FullName);
+                        AddToNamesInXamlIfRequired(field.FieldType);
                     }
-                    foreach (PropertyDefinition prop in type.Properties)
+                    foreach (var prop in type.Properties.Where(prop => prop.IsPublicOrInternal()))
                     {
-                        TypeDefinition propType;
-                        if (prop.IsPublic() && (propType = prop.PropertyType.Resolve()) != null && propType.IsEnum)
-                            namesInXaml.Add(prop.PropertyType.FullName);
+                        AddToNamesInXamlIfRequired(prop.PropertyType);
                     }
-                    foreach (MethodDefinition method in type.Methods)
-                    {
-                        TypeDefinition returnType;
-                        if (method.IsPublic() && (returnType = method.ReturnType.Resolve()) != null && returnType.IsEnum)
-                            namesInXaml.Add(method.ReturnType.FullName);
-                    }
-                }
-
-                // Extend with enums used in public properties on types implementing INotifyPropertyChanged
-                foreach (var type in info.GetAllTypeDefinitions().
-                    Where(t => t.ImplementsInterface("System.ComponentModel.INotifyPropertyChanged")))
-                {
-                    foreach (PropertyDefinition prop in type.Properties)
+                    foreach (var method in type.Methods.Where(method => method.IsPublicOrInternal()))
                     {
-                        TypeDefinition propType;
-                        if (prop.IsPublic() && (propType = prop.PropertyType.Resolve()) != null && propType.IsEnum)
-                            namesInXaml.Add(prop.PropertyType.FullName);
+                        AddToNamesInXamlIfRequired(method.ReturnType);
                     }
                 }
             }
 
-            return namesInXaml;
+            // Extend with enum property types which are being used in any relations to types implementing INotifyPropertyChanged
+            bool PublicOrInternalEnumPropertyFilter(PropertyDefinition property) =>
+                property.IsPublicOrInternal() && (property.PropertyType.Resolve()?.IsEnum ?? false);
+            var allTypeNamesRelatedToNpc = CollectTypesRelatedToNpc();
+            var allEnumTypeNamesRelatedToNpc = CollectAllTypes()
+                .SelectMany(type => type.Properties.Where(PublicOrInternalEnumPropertyFilter))
+                .Where(property => allTypeNamesRelatedToNpc.Contains(property.DeclaringType.FullName))
+                .Select(property => property.PropertyType.Resolve().FullName);

```

Comment:
Nice observed but no it is no longer needed as it is done in the `PublicOrInternalEnumPropertyFilter` check.
Hmm It will cost nothing so I will add it to make sure we do not run into issues with it in the future if the check is left out of the `PublicOrInternalEnumPropertyFilter` check.

Reply:
1eeadd344f7ddc44ef90cd6ce944d11c9cce88e6